constraint/pkg/externaldata/cache.go

```diff
@@ -0,0 +1,51 @@
+package externaldata
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/open-policy-agent/frameworks/constraint/pkg/apis/externaldata/v1alpha1"
+)
+
+type ProviderCache struct {
+	cache map[string]v1alpha1.Provider
+	mux   sync.RWMutex
+}
+
+func NewCache() *ProviderCache {
+	return &ProviderCache{
+		cache: make(map[string]v1alpha1.Provider),
+	}
+}
+
+func (c *ProviderCache) Get(key string) (v1alpha1.Provider, error) {
+	if v, ok := c.cache[key]; ok {
+		return v, nil
+	}
+	return v1alpha1.Provider{}, fmt.Errorf("key is not found in provider cache")
+}
+
+func (c *ProviderCache) Upsert(provider *v1alpha1.Provider) error {
```

Comment:
Do you foresee adding validation code? If so, returning an error here may make sense.

Reply:
added